Dockerfile

```diff
@@ -43,7 +44,8 @@ RUN \
   # allow anyone to connect by default
   sed -ie 's/127.0.0.1/%/' /usr/share/mariadb/mysql_system_tables_data.sql && \
   mkdir /run/mysqld && \
-  chown mysql:mysql /etc/my.cnf.d/ /run/mysqld /usr/share/mariadb/mysql_system_tables_data.sql
+  chown mysql:mysql /etc/my.cnf.d/ /run/mysqld /usr/share/mariadb/mysql_system_tables_data.sql && \
+  rm /etc/my.cnf.d/* -rf
```

Comment:
Is it necessary to remove this content? The idea is that you lean on mariadb defaults or mount your own config folder. I can see how it could be problematic if both configs are loaded; is that perhaps whats happening for you?

Reply:
#63 showed in my case.  I found both original `mariadb-server.cnf`  and `my.cnf` (with copied from `/tmp/my.cnf`). Then mysqld started with `port  0`
seems the default `cnf-file` will impact the initialization process.